api/lib/infrastructure/repositories/certification-candidate-repository.js

```diff
@@ -1,29 +1,48 @@
+const _ = require('lodash');
+const Bookshelf = require('../bookshelf');
 const CertificationCandidateBookshelf = require('../data/certification-candidate');
 const bookshelfToDomainConverter = require('../../infrastructure/utils/bookshelf-to-domain-converter');
+const { PGSQL_UNIQUE_CONSTRAINT_VIOLATION_ERROR } = require('../../../db/pgsql-errors');
 const {
   NotFoundError,
   CertificationCandidateCreationOrUpdateError,
   CertificationCandidateDeletionError,
   CertificationCandidateMultipleUserLinksWithinSessionError,
 } = require('../../domain/errors');
-const _ = require('lodash');
-const Bookshelf = require('../bookshelf');
-const { PGSQL_UNIQUE_CONSTRAINT_VIOLATION_ERROR } = require('../../../db/pgsql-errors');
 
 module.exports = {
 
   save(certificationCandidateToSave) {
-    const certificationCandidateBookshelf = new CertificationCandidateBookshelf(_.omit(certificationCandidateToSave, ['createdAt']));
+    const certificationCandidateBookshelf = new CertificationCandidateBookshelf(_adaptModelToDb(certificationCandidateToSave));
+
     return certificationCandidateBookshelf.save()
       .then((savedCertificationCandidate) => bookshelfToDomainConverter.buildDomainObject(CertificationCandidateBookshelf, savedCertificationCandidate))
       .catch((bookshelfError) => {
         if (bookshelfError.code === PGSQL_UNIQUE_CONSTRAINT_VIOLATION_ERROR) {
           throw new CertificationCandidateMultipleUserLinksWithinSessionError('A user cannot be linked to several certification candidates within the same session');
         }
+
         throw new CertificationCandidateCreationOrUpdateError('An error occurred while saving the certification candidate');
       });
   },
 
+  saveAll(certificationCandidates) {
+    return Promise.all(
+      certificationCandidates.map((certificationCandidate) =>
+        new CertificationCandidateBookshelf(_adaptModelToDb(certificationCandidate))
+          .save()
+          .then((result) => bookshelfToDomainConverter.buildDomainObject(CertificationCandidateBookshelf, result))
+          .catch((bookshelfError) => {
+            if (bookshelfError.code === PGSQL_UNIQUE_CONSTRAINT_VIOLATION_ERROR) {
+              throw new CertificationCandidateMultipleUserLinksWithinSessionError('A user cannot be linked to several certification candidates within the same session');
```

Comment:
Je connais pas trop la différence mais 'In' au lieu de "Within" dans `CertificationCandidateMultipleUserLinksWithinSessionError` serait plus simple à comprendre et plus court non ? 
(le nom est un peu à rallonge mais il décrit bien la situation, j'ai pas trouvé mieux)

Reply:
C'est pas dans le scope de la PR mais on peut changer. On change @sroccaserra @laura-bergoens ?